### ISSUE TITLE
Zip Code settings SE field validation does not work properly

### ISSUE THREAD

**Issue by SocialEngineMarket** [SocialEngine/phpv4-issues#495]
As you already know, we’ve developed Zip Code Radius Search Plugin. We have taken seriously the analysis of zip codes of different countries and realized that as of now SE field validation does not work properly. For example, if you add Profile field with Zip Code type to User’s Profile, the user won’t be able to add Canadian zip code.

Our specialists have already fixed the issue, which you can find in this file:

“application/modules/Fields/Form/Element/ZipCode.php”

But this makes our module imperfect and as a result it has to overwrite standard SE file. We would greatly appreciate it if you consider transferring our code into standard SE functionality.

Please see the code we offer below:

class Fields_Form_Element_ZipCode extends Engine_Form_Element_Text

{

  public function init()

  {

```
$this->addValidator('Regex', true, array(

  '/^([0-9]{3,6})$|^([0-9]{5}-[0-9]{4})$|^([a-zA-Z0-9]{1,2}[0-9]{1,4})$|^([a-zA-Z0-9]{2,4}\s[a-zA-Z0-9]{2,4})$|^([a-zA-Z0-9]{1,3}-[a-zA-Z0-9]{2,4})$/'

));



// Fix messages

$this->getValidator('Regex')->setMessage("'%value%' is not a valid zip code.", 'regexNotMatch');
```

  }

}

Your specialists can consider this code with a view to its security and make sure we offer a right decision.

Thank you for your time and hope you will find this useful.

**Comment by Khushboo016** [comment]
Hi @SocialEngineMarket , We had already added your suggestion in our internal feature tracker and we'll discuss it with our team regarding its implementation in SE code.

Thanks!

**Comment by Elshara** [comment]
I think the zip code field should be changed to only be visible if the city, state and country fields are visible along with the location as one field inparticular. This way it is all grouped together nicely so when someone selects their country, the rest can then be selected from the selections in order of state, city and zip.
Example.
Adding profile question location.
Would be split into a group of tabs. All of which being codependent on the other.
So if you select Canad as your country, the term state would automatically be replaced with Province. Then the term city would be visible as is, but the term zip would be replaced with postal code.
Similarly if you were to enter United Kingdom as your country, you could select where in the UK you are from, England, Northern Ireland, Whales, Scotland etc. Then the term province would be replaced with County, and the term Zip would be replaced with post code.
Something to consider, think about.
Adding a map could be optional.

**Comment by caseynally** [comment]
On three of my sites, we only use zipcode information and do not ask any other location based information. So hiding the zipcode if no other location fields are displayed would cause harm to some people who do this. I think a proper suggestion would be some sort of way for each webmaster to select triggers for each field. (no matter what the field) so that we can set custom view/hidden rules. For example if I created a field called, "Best Spongebob character" then based on what the user might select among my predefined choices will trigger other fields to hide or display.

**Comment by caseynally** [comment]
oh... also as far as your comment about changing names of field labels to match a users country...... you can use the SE built in language setting to do this..... just have your registration form offer the option to select one and then the rest of your items can show the correct language labels.

**Comment by Elshara** [comment]
Well dependent field selectors isn't a thing in SE.

**Comment by shivani2910** [comment]
This has been implemented in 4.9.

**Comment by Elshara** [comment]
Awesome!
